### ISSUE TITLE
Unable to set persistence directory for Redis Stack

### ISSUE THREAD

**Issue by ruslaniv** [RediSearch/RediSearch#3873]
I'm trying to setup persistence for Redis Stack. I have this entry in the redis.conf file:
```shell
dir /backup
```
I deploy RSS via ansible and use these commands among others
```yaml
...
- name: Ensure /backup directory exists on the host
    file:
      path: /backup
      state: directory
      mode: '0777'
...
- name: Upload redis.conf
    template: src=templates/redis.conf dest=/etc/redis/redis.conf
...
- name: Run redisearch container
    docker_container:
      name: redis-stack-server
      image: redis/redis-stack-server:latest
      volumes:
        - /backup:/backup:rw
...
```
Once the container is deployed I can log into the container and check that there is directory inside a container named `/backup` and if I create a file inside this directory, I can see this file both inside a redis container and on host machine.

Once I start indexing data into Redis, after 1_000 entries I'm getting an error
```shell
MISCONF Redis is configured to save RDB snapshots, but it's currently unable to persist to disk
```

When I check Redis logs I can see that:
```shell
14:M 19 Sep 2023 09:48:41.070 * Background saving started by pid 92
92:C 19 Sep 2023 09:48:41.071 # Failed opening the temp RDB file temp-92.rdb (in server root dir /) for saving: Permission denied
14:M 19 Sep 2023 09:48:41.174 # Background saving error
```
When I go to Redis CLI, I get this:
```shell
CONFIG GET dir
1) "dir"
2) "/"
```
although when I log into the container and print the `redis.conf` file, I can see that the `dir` directive is set to `/backup`

When I try to set the `dir` via Redis CLI, I get this error:
```shell
CONFIG SET dir "/backup"
(error) ERR CONFIG SET failed (possibly related to argument 'dir') - can't set protected config
localhost:6379> CONFIG SET dir ./
(error) ERR CONFIG SET failed (possibly related to argument 'dir') - can't set protected config
localhost:6379> CONFIG SET dir /data/tmp
(error) ERR CONFIG SET failed (possibly related to argument 'dir') - can't set protected config
```
 - OS: ubuntu 22.04
 - CPU model: Intel(R) Core(TM) i5-8500 CPU @ 3.00GHz
 - Version/branch: Redis server v=7.2.1
=============================

1. Why Redis is not reading the configuration file? 
2. Why  Redis is not setting the dir configuration during runtime from cli? 
3. How can I set the backup directory for Redis Stack?

**Comment by ruslaniv** [comment]
@alonre24 Thank you for your answer!
Indeed I was able to alter the `dir` settings from Redis cli after setting `enable-protected-configs` to `yes` in `redis.conf`
After that I indexed some data into redis and it did create an `rdb` file in `/backup`. 

**But the problem is** that for whatever reason redis still is not reading the persistence location from `redis.conf` 
Once I restarted the redis server and then checked the `CONFIG GET dir` I again got:
```shell
1) "dir"
2) "/"
```
So naturally the database was empty, it did not restore the backup since there was no backup file in `/` (it was in `/backup`) and when I tried indexing more data I was back to receiving the same error of redis not being not able to write to `/`

So why is redis not reading this setting from `redis.conf`? 

I bashed into redis container and I can see that there is a correct `redis.conf` file in `/etc/redis.conf` with all my settings like `dir /backup` but redis is not respecting this particular setting. 

====

> Also, note that for general redis issues, it is better to open an issue in [redis github repo](https://github.com/redis/redis).

Should I recreate the issue there?
